src/pure/utils.ts

```diff
@@ -80,31 +81,73 @@ export interface Cmd {
   args: string[]
 }
 
-export async function getVitestVersion(
-  vitestCommand?: Cmd,
-  env?: Record<string, string>,
-): Promise<string> {
-  let child
-  if (vitestCommand == null) {
-    child = spawn('npx', ['vitest', '-v'], {
-      stdio: ['ignore', 'pipe', 'pipe'],
-      env: { ...process.env, ...env },
-    })
-  }
-  else {
-    child = spawn(process.execPath, [vitestCommand.cmd, ...vitestCommand.args, '-v'], {
-      stdio: ['ignore', 'pipe', 'pipe'],
-      env: { ...process.env, ...env },
-    })
-  }
+/**
+ * Try to extract a vitest version from the output of the command
+ *
+ * @returns the version, or undefined if not found
+ */
+export const spawnVitestVersion = async (
+  command: string,
+  args: string[],
+  env?: Record<string, string | undefined>,
+): Promise<string | undefined> => {
+  log.info(`Trying to get vitest version from ${command} ${args.join(' ')}...`)
+
+  const child = spawn(command, args, {
+    stdio: ['ignore', 'pipe', 'pipe'],
+    env,
+  })
+
+  child.on('error', () => {
+    log.info('Command not found')
+  })
 
   // eslint-disable-next-line no-unreachable-loop
   for await (const line of chunksToLinesAsync(child.stdout)) {
```

Comment:
Maybe you should use try catch to wrap 106~114 here? Otherwise, it may just throw on 127 or 133 line

Reply:
Is `chunksToLinesAsync` throwing if the `spawn` fails?

When testing locally, if the spawn fails, we're not entering in the `for` loop and `detectVitestVersion` run the second command.

Reply:
I was afraid of the for await might throw. If you've tested it, that's fine. Thanks!

Reply:
Thank you for the review and the merge.

I'll add some tests, so we'll be able to test the edge cases if some new bugs are open, I found a lib to mock `spawn`